readers/timescale/messages.go

```diff
@@ -42,8 +42,9 @@ func (tr timescaleRepository) ReadAll(chanID string, rpm readers.PageMetadata) (
 
 	// If aggregation is provided, add time_bucket and aggregation to the query
 	if rpm.Aggregation != "" {
-		q = fmt.Sprintf(`SELECT EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000))) *1000000 AS time, publisher, protocol, subtopic, name, unit, %s(value) AS value FROM %s WHERE %s GROUP BY time, publisher, protocol, subtopic, name, unit ORDER BY time DESC LIMIT :limit OFFSET :offset;`, rpm.Interval, rpm.Aggregation, format, fmtCondition(rpm))
-		totalQuery = fmt.Sprintf(`SELECT COUNT(*) FROM (SELECT EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000))) AS time, %s(value) AS value FROM %s WHERE %s GROUP BY time) AS subquery;`, rpm.Interval, rpm.Aggregation, format, fmtCondition(rpm))
+		q = fmt.Sprintf(`SELECT EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000000))) *1000000000 AS time, %s(value) AS value, FIRST(publisher, time) AS publisher, FIRST(protocol, time) AS protocol, FIRST(subtopic, time) AS subtopic, FIRST(name,time) AS name, FIRST(unit, time) AS unit FROM %s WHERE %s GROUP BY 1 ORDER BY time DESC LIMIT :limit OFFSET :offset;`, rpm.Interval, rpm.Aggregation, format, fmtCondition(rpm))
```

Comment:
Do we need to use the `FIRST` statement?

Reply:
We do neeed it since the query without it results in this response in a 24 hour time period which is wrong: 
``` magistrala=# SELECT EXTRACT(epoch FROM time_bucket('24h', to_timestamp(time/1000000000))) *1000000000 AS bucketed_time, publisher, protocol, subtopic, name, unit, MIN(value) AS value FROM messages WHERE time >= 1722508942000000000 AND time <= 1722524662000000000 GROUP BY bucketed_time, publisher, protocol, subtopic, name, unit ORDER BY bucketed_time DESC LIMIT 150;
       bucketed_time        |              publisher               | protocol | subtopic |    name     | unit |         value          
----------------------------+--------------------------------------+----------+----------+-------------+------+------------------------
 1722470400000000000.000000 | 204e108e-4ae6-40bc-8fb1-3fa978db3191 | coap     |          | demovoltage | V    |  7.803559710230892e+17
 1722470400000000000.000000 | 5a9579bf-1fef-4f1d-ba68-0234b8f985a0 | coap     |          | demovoltage | V    |  2.412600813757681e+17
 1722470400000000000.000000 | 894301b9-2475-4f63-ba33-bd807998952f | coap     |          | demovoltage | V    |  4.024268556060997e+17
 1722470400000000000.000000 | 9f017d2f-73fd-4242-accc-aa692d3a4ba7 | coap     |          | demovoltage | V    |   4.45884579918682e+16
 1722470400000000000.000000 | a9bbbf86-27d6-470a-8bbf-a37fdd21b40c | coap     |          | demovoltage | V    | 2.0926923380874426e+17
 1722470400000000000.000000 | c2bd220b-1589-4b3d-8d96-e6dba857d27d | coap     |          | demovoltage | V    |   3.57573467897865e+17   
 1722470400000000000.000000 | c8b63d7a-b045-450f-af15-308c1ccc2c19 | coap     |          | demovoltage | V    | 1.0692956307192781e+17   
 1722470400000000000.000000 | dfd503af-f314-4e10-b295-427c57a23308 | coap     |          | demovoltage | V    | 4.4900347907806106e+17   
 1722470400000000000.000000 | dfdf1a24-1f3e-4fe6-bcc0-08953a2361a0 | coap     |          | demovoltage | V    | 5.2031105227597946e+17   
 1722470400000000000.000000 | f235c633-3b91-4e15-b481-c64a22925e34 | coap     |          | demovoltage | V    |  3.198428137180215e+17   
(10 rows)
```